File: bigdata/src/main/java/com/hl/bigdata/flink/stream/scala/source/MySourceWorld.scala
package com.hl.bigdata.flink.stream.scala.source

import org.apache.flink.streaming.api.functions.source.{RichParallelSourceFunction, SourceFunction}

/**
 * @author huanglin
 * @date 2024/07/03 17:20
 */
class MySourceWorld extends RichParallelSourceFunction[String]{

  override def run(sourceContext: SourceFunction.SourceContext[String]): Unit = {
    while(true) {
      sourceContext.collect("aa bb cc dd ee")
      println("产生一条数据")
      Thread.sleep(1000)
    }
  }

  override def cancel(): Unit = ???
}

File: bigdata/src/main/java/com/hl/bigdata/flink/stream/java/source/MySourceWorld.java
package com.hl.bigdata.flink.stream.java.source;

import org.apache.flink.streaming.api.functions.source.RichSourceFunction;

/**
 * @author huanglin
 * @date 2024/07/03 11:49
 */
public class MySourceWorld extends RichSourceFunction<String> {

    @Override
    public void run(SourceContext<String> sourceContext) throws Exception {
        while(true) {
            String world = "aa bb cc dd ";
            sourceContext.collect(world);
            Thread.sleep(1000);
        }
    }

    @Override
    public void cancel() {

    }
}
